src/parse/asp/builtins.go

```diff
@@ -753,6 +754,41 @@ func addDep(s *scope, args []pyObject) pyObject {
 	return None
 }
 
+func addDataToTargetAndMaybeQueue(s *scope, str string, target *core.BuildTarget) {
+	data := core.ParseBuildLabelContext(fmt.Sprint(str), s.pkg)
+	target.AddDatum(target.Label)
+	// Queue this dependency if it'll be needed.
+	if target.State() > core.Inactive {
+		err := s.state.QueueTarget(data, target.Label, true, false)
+		s.Assert(err == nil, "%s", err)
+	}
+}
+
+// Add runtime dependencies to target
+func addData(s *scope, args []pyObject) pyObject {
+	s.Assert(s.Callback, "can only be called from a pre- or post-build callback")
+	target := getTargetPost(s, string(args[0].(pyString)))
+
+	// add_data() builtin can take a string, list, or dict
+	if isType(args[1], "string") {
+		addDataToTargetAndMaybeQueue(s, fmt.Sprint(args[1].(pyString)), target)
+	} else if isType(args[1], "list") {
+		for str := range args[1].(pyList) {
```

Comment:
str here is going to be the index of the slice. You probably want:
```
for _, str := range args[1].(pyList) {}
```

Reply:
Thanks for pointing out the issue with str as well. Doesn't make the test pass though - my latest test is just adding a string so it should go into the first condition, it just ends up not adding the data for some reason